rmgpy/rmgobject.pyx

```diff
@@ -133,3 +134,52 @@ cpdef expand_to_dictionaries(obj):
             return expand_to_dictionaries(obj.as_dict())
         else:
             return obj
+
+cpdef recursive_make_object(obj, class_dictionary, make_final_object=True):
+    if isinstance(obj, dict):
+        if 'class' in obj:  # This is a dictionary of an object to be created
+            try:
+                class_to_make = class_dictionary[obj['class']]
+            except KeyError:
+                raise KeyError('Class {0} must be provided in the class_dictionary: {1}'.format(obj['class'],
+                                                                                                class_dictionary))
+
+            for key, val in obj.iteritems():
+                if key == 'class':
+                    continue
+                obj[key] = recursive_make_object(val, class_dictionary)
+
+            args = {key:obj[key] for key in obj if key != 'class'}
+            if make_final_object:
+                if hasattr(class_to_make, 'make_object'):
+                    created_obj = class_to_make.__new__(class_to_make)
+                    created_obj.make_object(args, class_dictionary)
+                else:
+                    created_obj = class_to_make(**args)
+                return created_obj
+            else:
+                return args
+        else:
+            for key, val in obj.iteritems():
+                obj[key] = recursive_make_object(val, class_dictionary)
+            return obj
+
+    elif isinstance(obj, list):
+        return [recursive_make_object(x, class_dictionary) for x in obj]
+
+    elif isinstance(obj, str):
```

Comment:
Can you do `isinstance(obj, (str, unicode))` instead? In Py3 (and Py-3 compatible code like ARC) all strings are unicode by default

Reply:
Done, thanks for the suggestion!

Reply:
In Py3, there is no unicode class, rather the default str class is unicode. So when we transition, I think we would need to remove the check for unicode? Is this correct?